src/Swarm/Doc/Gen.hs

```diff
@@ -419,6 +420,11 @@ recipeTable a rs = T.unlines $ header <> map (listToRow mw) recipeRows
 recipePage :: PageAddress -> [Recipe Entity] -> Text
 recipePage = recipeTable
 
+getBaseRobot :: Scenario -> Either Text Robot
```

Comment:
```suggestion
getBaseRobot :: Has (Throw SystemFailure) sig m => Scenario -> m Robot
```
Should be able to use this signature, then in the `Nothing` case you can `throwError $ CustomFailure "Scenario...`.  Then you don't need the `either (throwError . CustomFailure) return` below in `generateRecipe` at all.